apps/web/components/eventtype/IntervalLimitsManager.tsx

```diff
@@ -0,0 +1,166 @@
+import { Button, Select, TextField } from "@calcom/ui";
+import { FiPlus, FiTrash } from "@calcom/ui/components/icon";
+
+import type { IntervalLimitsKey } from "@lib/types/intervalLimitsKey";
+import { INTERVAL_LIMIT_OPTIONS, intervalOrderKeys } from "@lib/types/intervalLimitsKey";
+
+import type { FormValues } from "../../pages/event-types/[type]";
+
+type IntervalLimitItemProps = {
+  key: Key;
+  limitKey: IntervalLimitsKey;
+  step: number;
+  value: number;
+  textFieldSuffix?: string;
+  selectOptions: { value: keyof IntervalLimit; label: string }[];
+  hasDeleteButton?: boolean;
+  onDelete: (intervalLimitsKey: IntervalLimitsKey) => void;
+  onLimitChange: (intervalLimitsKey: IntervalLimitsKey, limit: number) => void;
+  onIntervalSelect: (interval: SingleValue<{ value: keyof IntervalLimit; label: string }>) => void;
+};
+
+const IntervalLimitItem = ({
+  limitKey,
+  step,
+  value,
+  textFieldSuffix,
+  selectOptions,
+  hasDeleteButton,
+  onDelete,
+  onLimitChange,
+  onIntervalSelect,
+}: IntervalLimitItemProps) => {
+  return (
+    <div className="mb-2 flex items-center space-x-2 text-sm rtl:space-x-reverse" key={limitKey}>
+      <TextField
+        required
+        type="number"
+        containerClassName={`${textFieldSuffix ? "w-36" : "w-16"} -mb-1`}
+        placeholder={`${value}`}
+        min={step}
+        step={step}
+        defaultValue={value}
+        addOnSuffix={textFieldSuffix}
+        onChange={(e) => onLimitChange(limitKey, parseInt(e.target.value))}
+      />
+      <Select
+        options={selectOptions}
+        isSearchable={false}
+        defaultValue={INTERVAL_LIMIT_OPTIONS.find((option) => option.value === limitKey)}
+        onChange={onIntervalSelect}
+      />
+      {hasDeleteButton && (
+        <Button variant="icon" StartIcon={FiTrash} color="destructive" onClick={() => onDelete(limitKey)} />
+      )}
+    </div>
+  );
+};
+
+type IntervalLimitsManagerProps = {
+  propertyName: "durationLimits" | "bookingLimits";
+  defaultLimit: number;
+  step: number;
+  textFieldSuffix?: string;
+};
+
+export const IntervalLimitsManager = ({
+  propertyName,
+  defaultLimit,
+  step,
+  textFieldSuffix,
+}: IntervalLimitsManagerProps) => {
+  const { watch, setValue, control } = useFormContext<FormValues>();
+  const watchIntervalLimits = watch(propertyName);
+  const { t } = useLocale();
+
+  const [animateRef] = useAutoAnimate<HTMLUListElement>();
+
+  return (
+    <Controller
+      name={propertyName}
+      control={control}
+      render={({ field: { value, onChange } }) => {
+        const currentIntervalLimits = value;
+
+        const addLimit = () => {
+          if (!currentIntervalLimits || !watchIntervalLimits) return;
+          const currentKeys = Object.keys(watchIntervalLimits);
+
+          const rest = Object.values(INTERVAL_LIMIT_OPTIONS).filter(
+            (option) => !currentKeys.includes(option.value)
+          );
+          if (!rest || !currentKeys) return;
```

Comment:
Empty arrays will always be truthy. This check is not guarding anything. If we're looking for empty arrays we should do this:

```suggestion
          if (!rest.length || !currentKeys.length) return;
```